internal/aws/shield_test.go

```diff
@@ -0,0 +1,177 @@
+package aws
+
+import (
+	"context"
+
+	"errors"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/shield"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloud_GetSubscriptionStatus(t *testing.T) {
```

Comment:
Tests in general look good, but the things I pointed out earlier come from not handling unexpected errors. Might be worth to add a transient error to each scenario just to see flow goes as expected still